code/src/components/TodoList/TodoList.js

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useRef } from "react";
+import styled from "styled-components/macro";
+import { TodoListItem } from "./TodoListItem";
+import { AddButton } from "./AddButton";
+import { EmptyListButton } from "./EmptyListButton";
+import { useSelector } from "react-redux";
+import { FinishAllTasksButton } from "./FinishAllTaskButton";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faRectangleList } from "@fortawesome/pro-light-svg-icons";
+import { grayColor } from "../style/colors";
+
+export const TodoList = () => {
+  const scrollIntoView = useSelector((store) => store.screen.scrollIntoView);
+  const todos = useSelector((store) => store.todo.todos);
+  const ButtonsContainerRef = useRef();
+
+  // Filter the todos list after categories and if the todo is completed or not and
+  // then conditionally render the lists depending on category and if completed or not.
+  const activities = todos.filter(
+    (todo) => todo.category === "activity" && !todo.completed
+  );
+  const business = todos.filter(
+    (todo) => todo.category === "business" && !todo.completed
+  );
+  const family = todos.filter(
+    (todo) => todo.category === "family" && !todo.completed
+  );
+  const completed = todos.filter((todo) => todo.completed);
+
+  // Using a useEffect to recognise changes in the global state and then
+  // either scrollIntoView or not.
+  useEffect(() => {
+    if (scrollIntoView) {
+      ButtonsContainerRef.current.scrollIntoView();
+    }
```

Comment:
Curious, what does this scrollintoview actually do? I couldn't locate it when using the app.